### DISCUSSION TITLE
Ane NativeWebView . the Cyrillic problem

### DISCUSSION THREAD

**Discussion by skramovskiy** [airsdk/Adobe-Runtime-Support#2551] [General]
working with NativeWebView

if I load the page via File.applicationDirectory.resolvePath - there are no problems, everything is displayed well.
But if they try to assign data via loadString, then there is a problem with displaying the Cyrillic alphabet. Please tell me how to be, maybe someone has encountered...

**Comment by ajwfrost** [comment]
Hi

Which platform are you using for this? (and which sort of webview, if on Windows?) - and which AIR SDK version?

If you can give us a quick example of this it would be good; a test here using Windows + both forms of webview, and a cyrillic string put into a loadString call was working and displaying fine...

thanks

**Reply by skramovskiy** [reply]
air sdk 50.2.2.2
[Ane NativeWebView  ](https://airnativeextensions.com/extension/com.distriqt.NativeWebView)
windows 11
```
`var webView:WebView = NativeWebView.service.createWebView( viewPort ) ;


webView.addEventListener( NativeWebViewEvent.LOCATION_CHANGE, webView_locationChangeHandler );
webView.addEventListener( NativeWebViewEvent.COMPLETE, webView_completeHandler );
webView.addEventListener( NativeWebViewEvent.ERROR, webView_errorHandler );

var htmlString:String ='<html>Тестовые символы</html>'
webView.loadString( htmlString );				
				`
```

**Reply by ajwfrost** [reply]
Interesting -> can you please try your string as `'<!DOCTYPE html><html>Тестовые символы</html>'`?
Or equally `'<html><body>Тестовые символы</body></html>'` may work?

**Reply by skramovskiy** [reply]
![image](https://user-images.githubusercontent.com/98939255/227974657-2564fdb5-4e0d-4656-bc4d-18fcf5120a52.png)

**Reply by ajwfrost** [reply]
Okay .. so I had tried this using our original (IE/WebBrowser based) StageWebView implementation, and adding the doctype worked. The full/formal text perhaps should be `'<!DOCTYPE html><html><head><meta charset="utf-8"></head><body>Тестовые символы</body></html>'` in case that works. Otherwise @marchbold are you able to check what's happening under the hood within the NativeWebView ANE?

thanks

**Reply by skramovskiy** [reply]
Yes. StageWebView - ok. NativeWebView - don't ok!

**Reply by marchbold** [reply]
@skramovskiy Post an issue in the extensions repository and one of our developers will look into it:

https://github.com/distriqt/ANE-NativeWebView/issues/new

**Reply by atollontech** [reply]
Had similar issues, resolved by using encodeURI/decodeURI methods.
